### ISSUE TITLE
Installation errors on Mac and Windows

### ISSUE THREAD

**Issue by s3alfisc** [apoorvalal/synthlearners#2]
Hi @apoorvalal , 

When trying to install on Mac or Windows, I get installation errors: 

On Mac, I get 

```
ERROR: Could not find a version that satisfies the requirement pyensmallen (from synthlearners) (from versions: none)
```
while on Windows

```
ERROR: Could not find a version that satisfies the requirement pyensmallen (from synthlearners) (from versions: none)
ERROR: No matching distribution found for pyensmallen
```

On Pypi, there seem to be distributions for MAC(ARM) and Linux: [link](https://pypi.org/project/pyensmallen/#files). 

I can provide more details on the machines if you think it helps. 

No problems running anything on Linux, fwiw =)

**Comment by apoorvalal** [comment]
would appreciate more details on macos and windows versions.

**Comment by s3alfisc** [comment]
Here are reproducible CI runs on github machines: [link](https://github.com/s3alfisc/synthlearners/actions/runs/11901037569/job/33163176102). I can work this into a full PR if you are interested. 

My windows info: 
```
Python implementation: CPython
Python version       : 3.12.7
IPython version      : 8.29.0

Compiler    : MSC v.1941 64 bit (AMD64)
OS          : Windows
Release     : 11
Machine     : AMD64
Processor   : Intel64 Family 6 Model 142 Stepping 12, GenuineIntel
CPU cores   : 8
Architecture: 64bit
```

**Comment by s3alfisc** [comment]
Interestingly, the github action for mac [does not fail](https://github.com/s3alfisc/synthlearners/actions/runs/11901037569/job/33163561210).

**Comment by apoorvalal** [comment]
actually these installation issues are likely more relevant for the pyensmallen repo.

**Comment by s3alfisc** [comment]
Yep sorry, you're right! I'll close this and move the issue to the other repo

**Comment by apoorvalal** [comment]
turned out to be a build system issue for older macs ; closing since it is lower priority for now
